### PR TITLE
Refactor WashingtonPostCollection class to have inner classes for Document

### PR DESCRIPTION
**Pull request by borislin**
Issue #313: Refactored `WashingtonPostCollection` - `Document` is now an inner class of the collection class.

### CODE REVIEW

File: src/main/java/io/anserini/collection/WashingtonPostCollection.java [hunk 1]
```diff
@@ -53,4 +58,137 @@ public FileSegment(Path path) throws IOException {
...
+        //  published date. So we just simply log a warning and return null
+        //  here in case future data may bring up this issue
+        LOG.warn("No unique ID or published date for this record, ignored...");
+        return null;
+      }
+
+      id = recordObj.id();
+      published_date = recordObj.published_date();
+
+      if (recordObj.contents != null && recordObj.contents.isPresent()) {
```
Comment (Peilin-Yang) [inline src/main/java/io/anserini/collection/WashingtonPostCollection.java]: This block seems to be to verbose, can we make it simpler?

Comment (borislin) [reply src/main/java/io/anserini/collection/WashingtonPostCollection.java]: When a field such as `contents` is missing in the JSON string, `recordObj.contents` will be equal to `null` after deserialization. When the JSON string contains `{"contents" : null}`, `recordObj.contents` will be equal to `Optional.empty()` after deserialization.

That's why I've used `if (recordObj.contents != null && recordObj.contents.isPresent())`.

Comment (borislin) [reply src/main/java/io/anserini/collection/WashingtonPostCollection.java]: That makes sense. I'll do it now.

**Comment by borislin** [comment]
@Peilin-Yang are you happy with this PR now?

**Pull request merged**